library/std/src/io/buffered/switchwriter.rs

```diff
@@ -0,0 +1,115 @@
+use crate::fmt::Arguments;
+use crate::io::{self, buffered::LineWriterShim, BufWriter, IoSlice, Write};
+/// Different buffering modes a writer can use
+#[unstable(feature = "stdout_switchable_buffering", issue = "78515")]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum BufferMode {
```

Comment:
If this is part of the public interface, it might be good to mark this as `#[non_exhaustive]` so we can add different modes later.

Alternatively it could be an option to not expose this enum, and have `set_unbuffered()`, `set_line_buffered(..)`, etc. That might make it a bit easier to change things in the future without breaking anything.

Reply:
I'm just a bit worried that it makes it more difficult to make changes/additions later. For example, `.get_mode() == BufferMode::Block` is not easily fixable if a size or other property is added later to block buffering mode. `.is_buffered()` or `.is_block_buffered()` is easier to maintain when things change.

Reply:
I'm not saying it's not a problem, but I am saying that C hasn't really invented an addition to this scheme since it became accepted amongst various OS in ~1983.

Reply:
Added `#[non_exhaustive]`

PS, I was pleasantly surprised to discover that `#[non_exhaustive]` has no effect in the type's defining crate, which is a totally obvious design in retrospect.